include/aspect/material_model/visco_plastic.h

```diff
@@ -382,6 +383,16 @@ namespace aspect
         Rheology::DislocationCreep<dim> dislocation_creep;
         Rheology::FrankKamenetskii<dim> frank_kamenetskii_rheology;
 
+        /**
+         * Whether to include peierls creep in the constitutive formulation.
+         */
+        bool use_peierls_creep;
```

Comment:
Why is this not another viscosityscheme? Is it because it acts in addition to the other schemes? But is `composite` not also a composite scheme? And what happens for `composite` with added peierls creep? In that case the strain rate  should be distributed between the three different creep mechanisms I think. Maybe you discussed all that, but it is not obvious to me from the code alone.

Reply:
I did things in this manner to ensure backwards compatibility. If we had Peierls creep as one of the viscosity schemes, then we would then need to include in the compositing with diffusion and dislocation creep. While we could set the default values to something that gives a very high viscosity (ex: 1e50 Pa s) and thus barely changes the composite viscosity values, I still thought it would be better to have it separately. 

Indeed, I checked for a simple example that doing the composite of Peierls creep with a composite of diffusion/dislocation creep gives the same thing as compositing all three values together. However, I will make derive the general expressions directly to be sure.

Having the Peierls creep separate also always composting with the Frank Kamenetskii viscosity, rather than diffusion/dislocation.

However, I'm open to restructuring in another way if it makes sense to do so. Restructuring will need to occur when we do the proper compositing as in `diffusion_dislocation.cc`, but for now I thought it would be best to ensure backwards compatibility.

Reply:
I don't think any restructuring will be necessary for the compositing. We'll just need to add a strain_rate_and_derivatives function.

See #3759 for how I envisage this will work.

Reply:
I did the math to check:
"Indeed, I checked for a simple example that doing the composite of Peierls creep with a composite of diffusion/dislocation creep gives the same thing as compositing all three values together. However, I will derive the general expressions directly to be sure."
Yes, it works out exactly the same.

Reply:
Thanks for checking @mibillen! Then I am ok with adding it as it is done here, it is easier because this way it can be combined with any of the other viscosity schemes.